tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -168,7 +169,12 @@ static ClustersAndEntries MakeClusters(const std::vector<std::string> &treeNames
       while ((start = clusterIter()) < entries) {
          end = clusterIter.GetNextEntry();
          // Add the current file's offset to start and end to make them (chain) global
-         clusters.emplace_back(EntryCluster{start + offset, end + offset});
+         const auto currentStart = std::max(start + offset, startEntry);
+         const auto currentEnd = std::min(end + offset, endEntry);
+         if (currentStart < currentEnd)
+            clusters.emplace_back(EntryCluster{currentStart, currentEnd});
+         if (currentEnd == endEntry) // if the desired end is reached, stop reading further
+            break;
```

Comment:
In general I would like to see many more comments with concrete examples of how this algo is treating the entry numbers, cluster boundaries etc., similar to https://github.com/root-project/root/blob/250785e56ae92c9e070f89b6dfdc9e3d26f3d1f6/bindings/experimental/distrdf/python/DistRDF/Ranges.py#L309

Reply:
thanks, I totally agree :)